toothpick-runtime/src/main/java/toothpick/Toothpick.java

```diff
@@ -222,6 +242,7 @@ public static void inject(Object obj, Scope scope) {
    */
   private static void removeScopeAndChildrenFromMap(ScopeNode scope) {
     MAP_KEY_TO_SCOPE.remove(scope.getName());
+    ROOT_SCOPES.remove(scope.getName());
```

Comment:
This methods is meant to remove a subtree its head. So removing every element form the ROOT_SCOPES map does not make a lot of sense here.

I think it makes more sense to remove inside `closeScope`:
```java
public static void closeScope(Object name) {
    // we remove the scope first, so that other threads don't see it, and see the next snapshot of
    // the tree
    ScopeNode scope = (ScopeNode) MAP_KEY_TO_SCOPE.remove(name);
    if (scope != null) {
      ScopeNode parentScope = scope.getParentScope();
      if (parentScope != null) {
        parentScope.removeChild(scope);
      } else {
        ROOT_SCOPES.remove(name);    // <<<<<<< HERE <<<<<<<<<
        ConfigurationHolder.configuration.onScopeForestReset();
      }
      removeScopeAndChildrenFromMap(scope);
    }
  }
```